scrape/target.go

```diff
@@ -414,6 +418,21 @@ func PopulateLabels(lset labels.Labels, cfg *config.ScrapeConfig) (res, orig lab
 		lb.Set(model.AddressLabel, addr)
 	}
 
+	if noDefaultPort {
+		// If it's an address with a trailing default port and the
+		// no-default-scrape-port flag is present, remove the port.
+		switch port {
+		case "80":
+			if scheme == "http" {
```

Comment:
@roidelapluie should this accept only an explicit `http` scheme, whereas when adding the port above both an explicit `http` and a clear scheme are accepted? I'm leaning towards yes, but I'm unsure.

https://github.com/prometheus/prometheus/blob/ef22afafd3d2d4d509cc2ba43edcf1c1a60efe9f/scrape/target.go#L411

Reply:
I don't think scheme can be empty , but maybe it can with relabeling?